asterisk_bridge.py

```diff
@@ -228,6 +230,31 @@ async def _handle_stasis_start(self, event):
                     # get external media channel port and create a stream
                     rtp_stream_in = await self.rtp_server.create_stream(self.channels[original_channel_id]['rtp_source_port_in'])
                     rtp_stream_out = await self.rtp_server.create_stream(self.channels[original_channel_id]['rtp_source_port_out'])
+
+                    # Wait a moment for RTP association to happen
+                    await asyncio.sleep(0.1)
+
+                    # Check if Asterisk swapped the RTP ports by looking at the remote_addr
+                    # If stream_in receives from port_in, ports are NOT swapped
+                    # If stream_in receives from port_out, ports ARE swapped
+                    ports_are_swapped = False
+                    if rtp_stream_in.remote_addr:
+                        source_port = rtp_stream_in.remote_addr[1]
+                        if source_port == int(self.channels[original_channel_id]['rtp_source_port_out']):
+                            ports_are_swapped = True
+
+                    # If ports are swapped, we need to swap the speaker names too
+                    if ports_are_swapped:
+                        final_speaker_name_in = speaker_name_out
+                        final_speaker_number_in = speaker_number_out
+                        final_speaker_name_out = speaker_name_in
+                        final_speaker_number_out = speaker_number_in
+                    else:
+                        final_speaker_name_in = speaker_name_in
+                        final_speaker_number_in = speaker_number_in
+                        final_speaker_name_out = speaker_name_out
+                        final_speaker_number_out = speaker_number_out
+
```

Comment:
just assign channel here 
               speaker_name_in = self.channels[original_channel_id]['caller_name']
                speaker_number_in = self.channels[original_channel_id]['caller_number']
                speaker_name_out = self.channels[original_channel_id]['connected_name']
                speaker_number_out = self.channels[original_channel_id]['connected_number']

and swap if needed insted of doing it on line 222-226